frontend/src/components/ViewActivities/activityfns.test.js

```diff
@@ -0,0 +1,92 @@
+import * as activityFns from './activityfns.js';
+
+const ten = new Date(Date.UTC(2020, 4, 2, 10,  0));          // May 2, 2020 10:00
+const eleven = new Date(Date.UTC(2020, 4, 2, 11, 0));        // May 2, 2020 11:00
+const elevenThirty = new Date(Date.UTC(2020, 4, 2, 11, 30)); // May 2, 2020 11:30
+const twelve = new Date(Date.UTC(2020, 4, 2, 12, 0));        // May 2, 2020 12:00
+const one = new Date(Date.UTC(2020, 4, 2, 13, 0));           // May 2, 2020 13:00
+const may102pm = new Date(Date.UTC(2020, 4, 10, 14, 0)); // May 10, 2020 14:00
+const may014pm = new Date(Date.UTC(2020, 4, 1, 16, 0));   // May 1, 2020 16:00
+const may153am = new Date(Date.UTC(2020, 4, 15, 3, 0));   // May 15, 2020 3:00
+
+function createActivity(startTime, endTime){
+  return {'start_time': startTime, 'end_time': endTime};
+}
+
+describe('Same day activity compareActivities', () => {
+  const tenToTwelve = createActivity(ten, twelve);
+  const elevenToOne = createActivity(eleven, one);
+  const elevenToElevenThirty = createActivity(eleven, elevenThirty);
+  const tenToEleven = createActivity(ten, eleven);
+  const elevenToTwelve = createActivity(eleven, twelve);
+
+  test('Overlapping activities', () => {
+    expect(activityFns.compareActivities(elevenToOne, tenToTwelve)).toBe(1);
+  })
+
+  test('One activity completely during another', () => {
+    expect(activityFns.compareActivities(tenToTwelve, elevenToElevenThirty)).toBe(-1);
+  })
+
+  test('Activities with same start time', () => {
+    expect(activityFns.compareActivities(tenToEleven, tenToTwelve)).toBe(-1);
+  })
```

Comment:
This tests 1/2 of the code path where both activities have the same start time. I recommend adding another test with the order "tenToTwelve, tenToEleven" to fully test the rest of that code path.